addons/pos_restaurant/static/src/overrides/models/pos_store.js

```diff
@@ -60,53 +60,85 @@ patch(PosStore.prototype, {
+            .filter((l) => typeof l.id !== "number");
+        const lineIdByOrderId = localOrderlines.reduce((acc, curr) => {
+            if (!acc[curr.order_id.id]) {
+                acc[curr.order_id.id] = [];
+            }
+            acc[curr.order_id.id].push(curr.id);
+            return acc;
+        }, {});
 
-            ["config_id", "in", [...this.config.raw.trusted_config_ids, this.config.id]],
+        const orders = await this.data.searchRead("pos.order", [
+            ["session_id", "=", this.session.id],
             ["table_id", "in", data["table_ids"]],
-            ["session_id", "=", odoo.pos_session_id],
         ]);
-        for (const table of data["order_count"]) {
-            tableByIds[table.id].uiState.changeCount = table.changes;
-            tableByIds[table.id].uiState.orderCount = table.orders;
-            tableByIds[table.id].uiState.skipCount = table.skip_changes;
+        await this.data.read(
+            "pos.order.line",
+            orders.flatMap((o) => o.lines).map((l) => l.id),
+            ["qty"]
+        );
+        for (const [orderId, lineIds] of Object.entries(lineIdByOrderId)) {
+            const lines = this.models["pos.order.line"].readMany(lineIds);
+            for (const line of lines) {
+                line.update({ order_id: orderId });
+            }
         }
-        if (
-            this.selectedTable &&
-            data["table_ids"].includes(this.selectedTable.id) &&
-            this.selectedOrderUuid
-        ) {
-            const changes = this.getOrderChanges();
-            if (changes.nbrOfChanges > 0) {
-                // at this stage we are sure to be on a concurrent pos session
-                // that needs to update its data from the server
-                try {
-                    this.tableSyncing = true;
-                    await this.syncAllOrders({ cancel_table_notification: true });
-                } finally {
-                    this.tableSyncing = false;
+
+        let isDraftOrder = false;
+        for (const order of orders) {
+            if (order.state !== "draft") {
+                this.removePendingOrder(order);
+                continue;
+            } else {
+                this.addPendingOrder([order.id]);
+            }
+
+            const tableId = order.table_id?.id;
+            if (!tableId) {
+                continue;
+            }
+
+            const draftOrder = this.models["pos.order"].find(
+                (o) => o.table_id?.id === tableId && o.id !== order.id && o.state === "draft"
+            );
+
+            if (!draftOrder) {
+                continue;
+            }
+
+            for (const orphanLine of draftOrder.lines) {
+                const adoptingLine = order.lines.find((l) => l.can_be_merged_with(orphanLine));
+                if (adoptingLine && adoptingLine.id !== orphanLine.id) {
+                    adoptingLine.merge(orphanLine);
+                } else if (!adoptingLine) {
```

Comment:
it would thus give that if the suggestion above is possible
```suggestion
                } else {
```